### PR TITLE
[Cli fix] Adding error check on cli

### PR DESCRIPTION
**Pull request by healthydeve**
## Description 

Added an error check on the object responses returned.

## Test Plan 

CI / Locally
---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui/src/client_commands.rs
Comment (gegaowp) [inline crates/sui/src/client_commands.rs:884]: nit: remove?

File: crates/sui/src/client_commands.rs
Comment (gegaowp) [inline crates/sui/src/client_commands.rs:1449]: you can do sth like
```
let owner_type = obj.owner.map(|o| ... ).map_err(|e| ... )?;
```

**Review by gegaowp — APPROVED** [review]
agree to unblock, please address comments before merging.

**Comment by 666lcz** [comment]
@healthydeve could you make a separate PR to address @gegaowp 's comments?

**Pull request merged**
